index.js

```diff
@@ -1,9 +1,61 @@
-var path = require("path");
+const fs = require('fs');
+const path = require('path');
+const maxmind = require('maxmind');
```

Comment:
Hmm, I don't mind it, however feels that consumers of this library should have a freedom of choice which lookup module they use. Thoughts?

Reply:
[maxmind](https://www.npmjs.com/package/maxmind) is another @runk package and not an official package from MaxMind (the company), though the name might seem that way. This is why I agree — we should let people use other packages that can search `.mmmd` files with this library (whose job is then to just prepare the databases but not query them).

Reply:
Fair point. Perhaps the `geolite2.open` and `geolite2.openSync` methods could request a second argument, that would be a function to build a database reader from a database path?

Example (end-user code):

```js
(async () => {
	const geolite2 = require('geolite2');
	const maxmind = require('maxmind'); // Or any other reader lib
	const fs = require('fs');
	
	// Second argument is a function to build a db reader from the file path
	// geolite2 will automatically rebuild the reader with the provided function on database updates
	const lookup = await geolite2.open('GeoLite2-City', async path => {
		return await maxmind.open(path); // Or custom lib code
		// (the returned object will be wrapped by geolite2's self-updating Proxy, should work with any object)
	});

	// Synchronous version
	const syncLookup = geolite2.openSync('GeoLite2-City', path => {
		return new maxmind.Reader(fs.readFileSync(path));
	});

	console.log(lookup.get('8.8.8.8'));
	console.log(syncLookup.get('8.8.8.8'));
})();
```

What do you think of this?

Reply:
I'd personally leave this detail for consumers. You might emit a signal that database file has changed, and they'd have their own means of ensuring they refreshed an instance of their lookup client or whatever.

Reply:
I have thought of this too - basically only providing an EventEmitter.
However since refreshing the reader instance on found updates is required by the license, I figured most users would end up with something looking like this code so we might as well do it in the lib.

Would it be OK with you if we allowed both the API I described above, and an EventEmitter-based one for more in-depth performance optimization/database usage in apps that need it?

Reply:
I looked into this again and realized adding events didn't make much sense. I'm going to push an updated API & Readme, feel free to open a new comment if you dislike that solution.

Reply:
fafb2b0

Reply:
8b9fdcc Added the EventEmitter API, documented.